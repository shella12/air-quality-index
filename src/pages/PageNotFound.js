import { useEffect } from 'react';
import { MdOutlineKeyboardBackspace } from 'react-icons/md';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { resetState } from '../redux/airquality/airquality';

const PageNotFound = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(resetState());
  });
  return (
    <div className="page-not-found">
      <Link className="page-not-found-link" to="/">
        <MdOutlineKeyboardBackspace />
&nbsp;Go Back
      </Link>
      <h2 className="page-not-found-heading">Error 404: Page Not Found</h2>
      <p className="page-not-found-heading">Requested page is not available</p>
    </div>
  );
};

export default PageNotFound;
