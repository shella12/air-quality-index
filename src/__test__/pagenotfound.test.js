import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import PageNotFound from '../pages/PageNotFound';

describe('testing PageNotFound component', () => {
  it('should render the PageNotFound component', () => {
    const pageNotFound = render(
      <Provider store={store}>
        <BrowserRouter>
          <PageNotFound />
        </BrowserRouter>
      </Provider>,
    );
    expect(pageNotFound).toMatchSnapshot();
  });
});
