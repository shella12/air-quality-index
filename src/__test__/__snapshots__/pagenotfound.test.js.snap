// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing PageNotFound component should render the PageNotFound component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-not-found"
      >
        <a
          class="page-not-found-link"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21v-2z"
            />
          </svg>
           Go Back
        </a>
        <h2
          class="page-not-found-heading"
        >
          Error 404: Page Not Found
        </h2>
        <p
          class="page-not-found-heading"
        >
          Requested page is not available
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-not-found"
    >
      <a
        class="page-not-found-link"
        href="/"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21v-2z"
          />
        </svg>
         Go Back
      </a>
      <h2
        class="page-not-found-heading"
      >
        Error 404: Page Not Found
      </h2>
      <p
        class="page-not-found-heading"
      >
        Requested page is not available
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
